src/main/java/ca/corefacility/bioinformatics/irida/service/remote/ProjectSynchronizationService.java

```diff
@@ -465,6 +488,39 @@ public void syncSingleEndSequenceFile(SingleEndSequenceFile file, Sample sample)
 		}
 	}
 
+	/**
+	 * Synchronize a given {@link Fast5Object} to the local
+	 * installation
+	 *
+	 * @param fast5Object
+	 *            the {@link Fast5Object} to sync
+	 * @param sample
+	 *            the {@link Sample} to add the file to
+	 */
+	public void syncFast5File(Fast5Object fast5Object, Sample sample) {
+		RemoteStatus fileStatus = fast5Object.getRemoteStatus();
+		fileStatus.setSyncStatus(SyncStatus.UPDATING);
+		try {
+			fast5Object = fast5ObjectRemoteService.mirrorSequencingObject(fast5Object);
+
+			fast5Object.setProcessingState(SequencingObject.ProcessingState.UNPROCESSED);
+			fast5Object.setFileProcessor(null);
+
+			fast5Object.getFile().setId(null);
+			fast5Object.getRemoteStatus().setSyncStatus(SyncStatus.SYNCHRONIZED);
+
+			objectService.createSequencingObjectInSample(fast5Object, sample);
+
+			fileStatus.setSyncStatus(SyncStatus.SYNCHRONIZED);
+
+			objectService.updateRemoteStatus(fast5Object.getId(), fileStatus);
```

Comment:
This whole section is exactly the same for single end and fast5 files, and nearly the same for pairs.  Want to encapsulate it into a private method that can be called here and in those other file types?

Reply:
Yup that was a good idea. Added in ccf8d68de322f55c99d3283986c593dd49df391e